testplan/testing/base.py

```diff
@@ -1076,10 +1107,13 @@ def update_test_report(self):
             return
 
         if len(self.result.report):
-            raise ValueError(
-                "Cannot update test report,"
-                " it already has children: {}".format(self.result.report)
-            )
+            for suite in self.result.report:
+                if suite.name not in [
+                    member.value for member in ResourceHooks
+                ]:
+                    raise ValueError(
+                        f"Cannot update test report, it already has a children: {self.result.report}"
+                    )
```

Comment:
This block is one of the changes that worth a look. In a nutshell, I made resource hook labels an Enum and also made sure that the ProcessRunnerTest subclasses can deal with their presence when trying to update the report with output from the process.